huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go

```diff
@@ -0,0 +1,306 @@
+				ForceNew:    true,
+				Description: `Specifies the name of the consumer group.`,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z|%-_0-9]*$`),
+						"An instance name starts with a letter and can contain only letters, digits,"+
+							"vertical lines(|), percent sign(%), underscores (_), and hyphens (-)"),
+					validation.StringLenBetween(3, 64),
+				),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the description of the consumer group.`,
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Specifies the state of the consumer group.`,
+			},
+			"coordinator_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Specifies the coordinator id of the consumer group.`,
+			},
+			"lag": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Specifies the lag number of the consumer group.`,
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Specifies the created time of the consumer group.`,
+			},
+		},
+	}
+}
+
+func resourceDmsKafkaConsumerGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	// createKafkaConsumerGroup: create DMS Kafka consumer group
+	var (
+		createKafkaConsumerGroupHttpUrl = "v2/{project_id}/kafka/instances/{instance_id}/group"
+		createKafkaConsumerGroupProduct = "dmsv2"
+	)
+	createKafkaConsumerGroupClient, err := cfg.NewServiceClient(createKafkaConsumerGroupProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsKafkaConsumerGroup Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	createKafkaConsumerGroupPath := createKafkaConsumerGroupClient.Endpoint + createKafkaConsumerGroupHttpUrl
+	createKafkaConsumerGroupPath = strings.ReplaceAll(createKafkaConsumerGroupPath, "{project_id}",
+		createKafkaConsumerGroupClient.ProjectID)
+	createKafkaConsumerGroupPath = strings.ReplaceAll(createKafkaConsumerGroupPath, "{instance_id}", instanceID)
+
+	createKafkaConsumerGroupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createKafkaConsumerGroupOpt.JSONBody = utils.RemoveNil(buildCreateKafkaConsumerGroupBodyParams(d, cfg))
+	_, createErr := createKafkaConsumerGroupClient.Request("POST",
+		createKafkaConsumerGroupPath, &createKafkaConsumerGroupOpt)
+
+	if createErr != nil {
+		return diag.Errorf("error creating DmsKafkaConsumerGroup: %s", createErr)
+	}
+
+	d.SetId(instanceID + "/" + d.Get("name").(string))
+	return resourceDmsKafkaConsumerGroupRead(ctx, d, meta)
+}
+
+func buildCreateKafkaConsumerGroupBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"group_name": utils.ValueIngoreEmpty(d.Get("name")),
```

Comment:
```suggestion
		"group_name": d.Get("name"),
```
the param of name is required, so it is not needed to judge whethere the value should ignore

Reply:
done